src/core/reactor_backend.cc

```diff
@@ -1355,7 +1504,16 @@ class reactor_backend_uring final : public reactor_backend {
                 seastar_logger.error("Invalid operation for iocb: {}", req.opname());
                 abort();
         }
+
+#ifdef USE_PASSTHRU
+        if (req.opcode() != o::append) {
```

Comment:
One requirement for iouring passthru should be explained. The user_data which is passed via sqe and returned via cqe should be started with the nvme command structure. Legacy code uses use_data field to pass the pointer of upper layer's completion. 
We already tried to utilize inheritance, however, it was not working because the first 8 bytes of class are fixed for representing vtable. We didn't find a better style that satisfies passthru's requirement for the user_data field. 
Our current approach replaces user_data field to structure 'passthru_arg' which starts with its member 'cmd' that represents nvme io command. We placed io_completion to the second field of 'passthru_arg' to invoke proper completion.